lib/logflare/application.ex

```diff
@@ -90,89 +81,136 @@ defmodule Logflare.Application do
     grpc_creds = if ssl, do: GRPC.Credential.new(ssl: ssl)
     pool_size = Application.get_env(:logflare, Logflare.PubSub)[:pool_size]
 
-    [
-      Logflare.ErlSysMon,
-      {Task.Supervisor, name: Logflare.TaskSupervisor},
-      {Cluster.Supervisor, [topologies, [name: Logflare.ClusterSupervisor]]},
-      Logflare.Repo,
-      {Phoenix.PubSub, name: Logflare.PubSub, pool_size: pool_size},
-      # supervisor(LogflareTelemetry.Supervisor, []),
-      # Context Caches
-      ContextCache,
-      Users.Cache,
-      Sources.Cache,
-      Billing.Cache,
-      SourceSchemas.Cache,
-      PubSubRates.Cache,
-      Logs.LogEvents.Cache,
+    # set goth early in the supervision tree
+    conditional_children() ++
```

Comment:
Maybe it is just me, but I find `Enum.concat/1` here more readable than `++`.

Reply:
i think for smaller lists it would be alright, but for such a large list, the nesting of lists within lists would make it harder to read. i think its better for smaller lists though